Makefile

```diff
@@ -3,47 +3,54 @@
 PREFIX := /usr
 DESTDIR :=
 
-ifndef VERSION
-MAJORVERSION := 8
-MINORVERSION ?= $(shell git rev-list --count master)
-endif
-VERSION := ${MAJORVERSION}.${MINORVERSION}
+
+MAJORVERSION$(VERSION) := 8
+MINORVERSION$(VERSION) != git rev-list --count master
+VERSION ?= ${MAJORVERSION}.${MINORVERSION}
 
 LDFLAGS := -ldflags '-s -w -X main.version=${VERSION}'
 ARCH := $(shell uname -m)
 PKGNAME := yay
 BINNAME := yay
 PACKAGE := ${PKGNAME}_${VERSION}_${ARCH}
 
-export GOPATH=$(shell pwd)/.go
+GOPATH=$(shell pwd)/.go
 
 default: build
 
+.PHONY: all
 all: | clean package
 
+.PHONY: install
 install:
 	install -Dm755 ${BINNAME} $(DESTDIR)$(PREFIX)/bin/${BINNAME}
 	install -Dm644 doc/${PKGNAME}.8 $(DESTDIR)$(PREFIX)/share/man/man8/${PKGNAME}.8
 	install -Dm644 completions/bash $(DESTDIR)$(PREFIX)/share/bash-completion/completions/${PKGNAME}
 	install -Dm644 completions/zsh $(DESTDIR)$(PREFIX)/share/zsh/site-functions/_${PKGNAME}
 	install -Dm644 completions/fish $(DESTDIR)$(PREFIX)/share/fish/vendor_completions.d/${PKGNAME}.fish
 
+.PHONY: uninstall
 uninstall:
 	rm -f $(DESTDIR)$(PREFIX)/bin/${BINNAME}
 	rm -f $(DESTDIR)$(PREFIX)/share/man/man8/${PKGNAME}.8
 	rm -f $(DESTDIR)$(PREFIX)/share/bash-completion/completions/${PKGNAME}
 	rm -f $(DESTDIR)$(PREFIX)/share/zsh/site-functions/_${PKGNAME}
 	rm -f $(DESTDIR)$(PREFIX)/share/fish/vendor_completions.d/${PKGNAME}.fish
 
+.PHONY: test
 test:
 	gofmt -l *.go
 	@test -z "$$(gofmt -l *.go)" || (echo "Files need to be linted" && false)
 	go vet
 	go test -v
 
-build:
+.PHONY: build
+build: ${BINNAME}
+
+${BINNAME}:
```

Comment:
Prevent rebuild when the file already exists and is up to date. Really lots of `.PHONY:` rules is generally a bad sign in a makefile, I didn't want to touch too much though so I didn't extract out the generation of any other build artifacts except the main one.

Reply:
> Really lots of .PHONY: rules is generally a bad sign in a makefile

I find this true when using C and such. In this case though `go build` already knows when to rebuild and when not to.

Reply:
Also my preference is all the recipes under one .PHONY declaration. But that's about all I have to complain about as long as it builds.

Reply:
That's a fair point; it still feels poor to shell out unnecessarily if you don't have to, but I suppose it matters less in Go where the build is smart and fast.

Reply:
Oh gosh, was that there the whole time? I didn't even notice that they were already .PHONY, sorry about that. Fixing.

Reply:
Haha line 1 and all. Don't worry about it.